### PR TITLE
Audit Events Ingestion Spec update And Denorming Module Level Data

### PR DESCRIPTION
**Pull request by manjudr**
1. Audit Event Ingestion spec
2. Denorming the Module Level Data

### CODE REVIEW

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 1]
```diff
@@ -0,0 +1,680 @@
...
+      "type": "string",
+      "parseSpec": {
+        "format": "json",
+        "flattenSpec": {
+          "useFieldDiscovery": false,
+          "fields": [
+            {
+              "type": "root",
+              "name": "eid"
+            },
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:15]: eid is not required when we index audit events only

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 2]
```diff
@@ -0,0 +1,680 @@
...
+            {
+              "type": "path",
+              "name": "context_cdata_type",
+              "expr": "$.context.cdata[*].type"
+            },
+            {
+              "type": "path",
+              "name": "context_cdata_id",
+              "expr": "$.context.cdata[*].id"
+            },
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:60]: cdata should not be indexed

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+            {
+              "type": "path",
+              "name": "object_rollup_l2",
+              "expr": "$.object.rollup.l2"
+            },
+            {
+              "type": "path",
+              "name": "edata_props",
+              "expr": "$.edata.props[*]"
+            },
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:90]: props is not required to be indexed

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+    },
+    "transformSpec": {
+      "transforms": [
+        {
+          "type": "expression",
+          "name": "device_first_access",
+          "expression": "timestamp_format(device_first_access, 'YYYY-MM-dd','Asia/Kolkata')"
+        }
+      ]
+    },
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:646]: Not required

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 5]
```diff
@@ -0,0 +1,680 @@
...
+  },
+  "ioConfig": {
+    "topic": "{{env}}.telemetry.audit",
+    "consumerProperties": {
+      "bootstrap.servers": "{{kafka_brokers}}"
+    },
+    "taskCount": 1,
+    "replicas": 1,
+    "taskDuration": "PT14400S",
+    "useEarliestOffset": false
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:674]: it should be similar to telemetry_feedback_events spec. It should filter for audit events as below
"transformSpec": {
      "filter": {
        "type": "selector",
        "dimension": "eid",
        "value": "AUDIT"
      }
    }

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+              "type": "root",
+              "name": "eid"
+            },
+            {
+              "type": "path",
+              "name": "context_channel",
+              "expr": "$.context.channel"
+            },
+            {
+              "type": "path",
```
[the remark this answers is not in the document]
Comment (manjudr) [reply ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka]: Indexing content, collection, l2data, user, location denromed fields Santhosh.

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 7]
```diff
@@ -0,0 +1,650 @@
...
+    "granularitySpec": {
+      "type": "uniform",
+      "segmentGranularity": "DAY",
+      "queryGranularity": "DAY",
+      "rollup": true,
+      "intervals": null
+    }
+  },
+  "ioConfig": {
+    "topic": "{{env}}.telemetry.audit",
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:637]: Topic is still incorrect.

File: ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka [hunk 8]
```diff
@@ -0,0 +1,650 @@
...
+              "expr": "$.derivedlocationdata.district"
+            },
+            {
+              "type": "path",
+              "name": "derived_loc_from",
+              "expr": "$.derivedlocationdata.from"
+            },
+            {
+              "type": "path",
+              "name": "l2data_created_for",
```
Comment (SanthoshVasabhaktula) [inline ansible/roles/druid-ingestion/templates/telemetry_audit_rollup_syncts_kafka:299]: replace l2data with object_l2

**Pull request merged**